map/src/infoblock/components/graph/RoadAttributesGraphProvider.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useContext, useEffect, useMemo, useState } from 'react';
+import _ from 'lodash';
+import AppContext from '../../../context/AppContext';
+import { addDistanceToPoints, getTrackPoints } from '../../../context/TracksManager';
+import RoadAttributesGraph from './RoadAttributesGraph';
+
+export default function RoadAttributesGraphProvider({ width }) {
+    const ctx = useContext(AppContext);
+
+    const [roadPoints, setRoadPoints] = useState(null);
+
+    useEffect(() => {
+        if (ctx.selectedGpxFile && ctx.currentObjectType === ctx.OBJECT_TYPE_CLOUD_TRACK) {
+            let points = _.cloneDeep(getTrackPoints(ctx.selectedGpxFile));
+            if (points[0].segment) {
+                setRoadPoints(points);
```

Comment:
please think about calling setRoadPoints only if they were really changed (apply and test on GpxGraph too); good idea to add console.log at begin of useMemo's function to check how often it was called